File: Navtrack.Api/Models/DeviceModel.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Navtrack.Api.Models
{
    public class DeviceModel : IModel
    {
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; }
        
        [Required]
        public string DeviceId { get; set; }
        
        public int DeviceModelId { get; set; }
        
        [JsonPropertyName("deviceModel")]
        public DeviceModelModel Model { get; set; }
    }
}

File: Navtrack.Core/ClientApp/src/services/api/types/device/DeviceModel.ts
import { DeviceModelModel } from "./DeviceModelModel";

export type DeviceModel = {
  id: number;
  deviceId: string;
  name: string;
  deviceModelId: number;
  deviceModel?: DeviceModelModel;
};

export const DefaultDeviceModel: DeviceModel = {
  id: 0,
  deviceId: "",
  name: "",
  deviceModelId: 0
};
